core/src/test/scala/unit/kafka/server/ConsumerGroupHeartbeatRequestTest.scala

```diff
@@ -36,54 +35,68 @@ import scala.collection.Map
 import scala.jdk.CollectionConverters._
 
 @ExtendWith(value = Array(classOf[ClusterTestExtensions]))
-class ConsumerGroupHeartbeatRequestTest(cluster: ClusterInstance) {
-
-  @ClusterTest(
+@ClusterTestDefaults(
     types = Array(Type.KRAFT),
     serverProperties = Array(
-      new ClusterConfigProperty(key = GroupCoordinatorConfig.GROUP_COORDINATOR_REBALANCE_PROTOCOLS_CONFIG, value = "classic"),
       new ClusterConfigProperty(key = GroupCoordinatorConfig.OFFSETS_TOPIC_PARTITIONS_CONFIG, value = "1"),
       new ClusterConfigProperty(key = GroupCoordinatorConfig.OFFSETS_TOPIC_REPLICATION_FACTOR_CONFIG, value = "1")
     )
+)
+class ConsumerGroupHeartbeatRequestTest(cluster: ClusterInstance) {
+
+  @ClusterTests(value = Array(
+      new ClusterTest(
+          serverProperties = Array(new ClusterConfigProperty(key = GroupCoordinatorConfig.GROUP_COORDINATOR_REBALANCE_PROTOCOLS_CONFIG, value = "classic")),
+          tags = Array("0")
+      ),
+      new ClusterTest(
+          serverProperties = Array(new ClusterConfigProperty(key = GroupCoordinatorConfig.GROUP_COORDINATOR_REBALANCE_PROTOCOLS_CONFIG, value = "classic")),
+          tags = Array("1")
+      )
+    )
   )
```

Comment:
I’m currently testing different versions of `ConsumerGroupHeartbeatRequest` using the `tags` property as a workaround. It feels quite clunky, but I haven’t found a way to do parameterized tests within the cluster test context. 🥵

Reply:
Yeah, this is a bit too extreme. We have ApiVersionsSource but you cannot combine it with ClusterTest. Honestly, the best may be to just loop over the version within the test for now.

Reply:
I have tried using a `for` loop locally, but since this approach doesn’t provide a 'clean' environment for each iteration, the assertions might get polluted.
Is it acceptable ?

Reply:
For example:
in the `testStaticMemberRemovedAfterSessionTimeoutExpiryWhenNewGroupCoordinatorIsEnabled` test case，we validate the `memberEpoch`．If we use a `for` loop in this case，since we don't restart clusters, we would need to set the expected value in a tricky way．

Reply:
We don't need to parameterize all the existing tests. How about the following? We make all the existing tests use the last version (so they provided the member id). We add one test which ensure that version 0 generate the member id on the server side. We add one test which ensure that versions > 0 fails if the member id is not provided. This should give us enough coverage. What do you think?

Reply:
Thanks for feedback.
I think this approach is very elegant. 👍🏼